vncdotool/client.py

```diff
@@ -243,7 +244,15 @@ def _capture(self, filename, *args):
         self.deferred.addCallback(self._captureSave, filename, *args)
         return self.deferred
 
+    def framebufferUpdateRequest(self, *args):
+        self.screen_updated = False
+        return super(VNCDoToolClient, self).framebufferUpdateRequest(*args)
+
     def _captureSave(self, data, filename, *args):
+        if not self.screen_updated:
```

Comment:
Rather than introduce a new data member could we branch of `if not self.screen`?

Reply:
If we make the change suggested in commitUpdate, I think this change may be unnecessary.

Reply:
No, this is not just a problem when the screen hasn't been updated at all. It's a problem every time a screenshot is created, so you can't tell just by looking at the screen attribute whether or not it's been updated recently.

The goal of this is to always update the screen before taking a screenshot, so it's always up to date -- not just ensure we have something in the buffer, which could be arbitrarily old.